update: combine documentation mcp server and aiven mcp server docs pages [EVERSQL-1782] (#1476)
aiven/aiven-docs: docs/tools/mcp-server.md

---
title: Aiven MCP
sidebar_label: Aiven MCP
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import CodeBlock from '@theme/CodeBlock';
import {cursorDeepLink, mcpUrl} from '@site/src/components/mcpAivenLiveConstants';

> Aiven offers two MCP servers for use with AI assistants:

Reviewer: ```suggestion
Aiven offers two Model Context Protocol (MCP) servers for use with AI assistants:
```

- **[Aiven MCP server](#aiven-mcp-server)**: Create and manage Aiven services directly from AI assistants
  such as Cursor and Claude Code.
- **[Documentation MCP server](#documentation-mcp-server)**: Access Aiven documentation from MCP-compatible clients.

> 

Reviewer: Maybe we could add some guidance here on when to use each?

## Aiven MCP server

> Use the Aiven Model Context Protocol (MCP) server to create and manage Aiven services from

Reviewer: ```suggestion
Use the Aiven MCP server to create and manage Aiven services from
```

AI assistants, such as Cursor and Claude Code.

### Prerequisites

- An [Aiven account](https://console.aiven.io/signup)

> - An organization with MCP access enabled

Reviewer: Is this because the feature is still limited availability? How does one enable this access?

Author: Yes, during limited availability (next few weeks), users request access during authentication after install. Since this will be removed soon, I think we can omit it from prerequisites section, and the user will see it as part of the authentication process.

Reviewer: You can also add an LA tag by setting `limited: true` in the frontmatter ([example](https://github.com/aiven/aiven-docs/blob/main/docs/products/datahub.md?plain=1)). That just makes it clear that it's being developed/could change, and has different support terms.

- An MCP-compatible client, such as Cursor, Claude Code, Claude Desktop, or

>   Visual Studio Code

Reviewer: ```suggestion
  Visual Studio Code (VS Code)
```

> ### MCP server URL

Reviewer: It looks like this section is duplicating a detail already shown in the instructions for configuring - can we remove it?

Use the following server URL when configuring your client:

<CodeBlock language="text">{mcpUrl}</CodeBlock>

### Configure your MCP client {#configure-aiven-mcp}

<Tabs groupId="aiven-mcp-clients">
<TabItem value="cursor" label="Cursor" default>

<!-- vale off -->
<a
  href={cursorDeepLink}
  style={{display: 'inline-flex', alignItems: 'center', gap: '10px', padding: '10px 20px', backgroundColor: '#2d2d2d', color: 'white', borderRadius: '8px', textDecoration: 'none', fontSize: '16px', fontWeight: 500, border: 'none', lineHeight: '20px'}}
>
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="20"
    height="20"
    viewBox="0 0 24 24"
    fill="none"
    aria-hidden="true"
    focusable="false"
    style={{display: 'block', flex: 'none'}}
  >
    <path d="M21 7.5L12 2L3 7.5M21 7.5L12 13M21 7.5V16.5L12 22M3 7.5L12 13M3 7.5V16.5L12 22M12 13V22" stroke="currentColor" strokeWidth="2" strokeLinejoin="round" strokeLinecap="round"/>
  </svg>
  <span>Add to Cursor</span>
</a>
<!-- vale on -->

<br /><br />

To add the server manually:

> 1. In your project root, create or edit `.cursor/mcp.json`.

Reviewer: Nit: small change to align with style guide for talking about files
```suggestion
1. In your project root, create or edit the `.cursor/mcp.json` file.
```

1. Add the following configuration:

   <!-- vale off -->
   <CodeBlock language="json">{JSON.stringify({mcpServers: {aiven: {type: "http", url: mcpUrl}}}, null, 2)}</CodeBlock>
   <!-- vale on -->

1. Save the file.
1. Restart Cursor.
1. Open **Settings** > **Tools & MCP**.
1. Select **aiven** and click **Connect**.

> 

Reviewer: What do you think about also adding a link to the relevant Cursor doc here?

</TabItem>
<TabItem value="claude-code" label="Claude Code">

1. Open a terminal.
1. Run the following command:

   <CodeBlock language="bash">{`claude mcp add --transport http aiven ${mcpUrl}`}</CodeBlock>

> 1. Run `/mcp` in Claude Code to verify the server is registered.

Reviewer: Nit: we usually start with the reason "to [do thing]" in these cases
```suggestion
1. To verify the server is registered, in Claude Code run `/mcp`.
```

> 1. The first time you use the server, your browser opens.

Reviewer: I would put this sentence at the end of the previous step since it is a result of that rather than a step the user has to take.

1. Sign in to Aiven and select your organization.

For more information, see the [Claude Code MCP documentation](https://docs.anthropic.com/en/docs/claude-code/tutorials/set-up-mcp).

</TabItem>
<TabItem value="claude-desktop" label="Claude Desktop">

1. Open the Claude Desktop configuration file. If it does not exist, create it:

   - **macOS:**
     `~/Library/Application Support/Claude/claude_desktop_config.json`

   - **Windows:**
     `%APPDATA%\Claude\claude_desktop_config.json`

1. Add the following configuration to the file:

   <!-- vale off -->
   <CodeBlock language="json">{JSON.stringify({mcpServers: {aiven: {type: "http", url: mcpUrl}}}, null, 2)}</CodeBlock>
   <!-- vale on -->

1. Save the file and restart Claude Desktop.

For more information, see the [Claude Desktop MCP documentation](https://modelcontextprotocol.io/quickstart/user).

</TabItem>
<TabItem value="vscode" label="VS Code">

:::note

> Requires Visual Studio Code 1.102 or later with the GitHub Copilot extension installed

Reviewer: ```suggestion
Requires VS Code 1.102 or later with the GitHub Copilot extension installed
```

and enabled.
:::

> 1. Open your workspace in Visual Studio Code.

Reviewer: ```suggestion
1. Open your workspace in VS Code.
```

> 1. In the workspace root, create a `.vscode` directory if it does not exist.

Reviewer: ```suggestion
1. In the workspace root, create a `.vscode` directory.
```

> 1. In the `.vscode` directory, create or edit `mcp.json`.

Reviewer: ```suggestion
1. In the `.vscode` directory, create or edit the `mcp.json` file.
```

1. Add the following configuration:

   <!-- vale off -->
   <CodeBlock language="json">{JSON.stringify({servers: {aiven: {type: "http", url: mcpUrl}}}, null, 2)}</CodeBlock>
   <!-- vale on -->

1. Save the file.

> 1. Reload Visual Studio Code.

Reviewer: ```suggestion
1. Reload VS Code.
```

1. Open the Command Palette and run **MCP: List Servers**.
1. Confirm that **aiven** appears in the list.

For more information, see the [VS Code MCP documentation](https://code.visualstudio.com/docs/copilot/customization/mcp-servers).

</TabItem>
<TabItem value="other" label="Other clients">

1. Open your MCP client configuration.
1. Add the Aiven MCP server using the following URL:

   <CodeBlock language="text">{mcpUrl}</CodeBlock>

> Most clients use a configuration similar to:

Reviewer: This should also be indented, but maybe we don't the URL in a separate code block? It could be inline like "Add the Aiven MCP server using the URL {mcpUrl}. Most clients....". Just a thought. 

The indentation also affects the numbering. 

<img width="1008" height="441" alt="Image" src="https://github.com/user-attachments/assets/bf7e0ad8-7b28-4d29-aeef-47045f9d49bc" />

   check that **aiven** appears with a connected status.

</TabItem>
<TabItem value="claude-code" label="Claude Code">

1. Run `/mcp` in Claude Code to verify the server is registered.
1. Try a prompt such as:

   > List my Aiven projects.

1. If your client prompts you to allow tool execution, approve the request.

</TabItem>
<TabItem value="claude-desktop" label="Claude Desktop">

1. Open a new conversation.
1. Verify that the MCP tools icon appears in the input area.
1. Try a prompt such as:

   > List my Aiven projects.

1. If prompted to allow tool execution, click **Allow**.

</TabItem>
<TabItem value="vscode" label="VS Code">

1. Open Copilot Chat in Visual Studio Code.
1. Open the Command Palette and run **MCP: List Servers**.
1. Confirm that **aiven** appears in the list.
1. Try a prompt such as:

   > List my Aiven projects.

1. If prompted to allow tool execution, click **Allow**.

</TabItem>
<TabItem value="other" label="Other clients">

1. Open your AI assistant.
1. Try a prompt such as:

   > List my Aiven projects.

1. If your client prompts you to allow tool execution, approve the request.

> 1. Verify that the response includes information about your Aiven projects.

Reviewer: Nit: This step isn't included for the other tools and is fairly obvious.
```suggestion

```

</TabItem>
</Tabs>

> ### Authentication

Reviewer: I think this should be the first section (before the Prerequisites).

The Aiven MCP server uses OAuth 2.0 with Proof Key for Code Exchange (PKCE) for
authentication. The first time you use the server, your browser opens so you can sign in
to Aiven and select your organization. The server refreshes tokens automatically.

### Security and responsibility

> :::note

Reviewer: This seems like it warrants a stronger admonition title.
```suggestion
:::important
```

creating, modifying, and deleting services, databases, topics, and data. AI agents can
run operations from natural language prompts that are easy to misinterpret. Using the
Aiven MCP server can result in damage to or loss of data. Whether to enable MCP access
is your organization's decision. Evaluate the risks before you grant AI agents access
to your resources.
:::

Under the [shared responsibility model](https://aiven.io/responsibility-matrix),
security and compliance for MCP usage are shared between Aiven and your organization.
Aiven secures the platform and API. You are responsible for the following:

- **Deciding whether to enable MCP** in your organization and evaluating the
  associated risks.
- **Controlling access** by scoping API tokens to the minimum permissions needed
  (principle of least privilege) and rotating them regularly.
- **Reviewing AI agent actions** before they run, especially for write or delete
  operations on production resources.
- **Configuring MCP servers securely**, including choosing
  [read-only mode](/docs/tools/mcp-server#read-only-mode) where appropriate to
  restrict the server to non-destructive operations.

> ### Supported tools

Reviewer: I think this should be in the introduction and doesn't need to be a separate section. For example:

> Use the Aiven Model Context Protocol (MCP) server to create and manage Aiven services from AI assistants, such as Cursor and Claude Code. You can create, update, and delete Aiven services across all supported service types. It also supports browsing available plans and pricing, view service metrics and logs, and manage
service configurations and cloud regions. The supported services and features include:
- Aiven for PostgreSQL: Create and manage....
- Aiven for Apache Kafka: ...

#### Service management

Create, update, and delete Aiven services across all supported service types.
Browse available plans and pricing, view service metrics and logs, and manage
service configurations and cloud regions.

#### PostgreSQL®

Create and manage PostgreSQL databases, execute read and write SQL queries,
and manage PgBouncer connection pools. Get AI-powered query optimization
recommendations, view query performance statistics, and list available extensions.

#### Apache Kafka®

Create and manage Kafka topics, produce and consume messages, and configure
Kafka Connect connectors. Browse Schema Registry subjects and manage
connector lifecycle operations including pause, resume, and restart.

> ### Read-only mode

Reviewer: Could also be included in the intro, I think or at least brought up higher so that users know this is an option (given the potential risks with letting the agent make changes).
